### PR TITLE
Added code investigator code

### PR DESCRIPTION
**Pull request by Nazariii**

### CODE REVIEW

File: src/main/scala/com/crimes/CrimeInvestigatorHelper.scala [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+    *
+    * @param args program args
+    * @return directory name
+    */
+  def getDirectoryName(args: Array[String]): String = {
+    if (args.isDefinedAt(0)) {
+      args(0)
+    } else {
+      defaultDirectory
+    }
```
Comment (azhur) [inline src/main/scala/com/crimes/CrimeInvestigatorHelper.scala]: `args.headOption.getOrElse(defaultDirectory)` ;)

File: src/main/scala/com/crimes/CrimeInvestigatorHelper.scala [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+  }
+
+  /**
+    * Check if crime has CrimeId, latitude and longitude specified.
+    *
+    * @param crime to check
+    * @return true if CrimeId, latitude and longitude specified.
+    */
+  def isIdOrLocationIsNotEmpty(crime: Crime): Boolean = {
```
Comment (azhur) [inline src/main/scala/com/crimes/CrimeInvestigatorHelper.scala]: maybe just `isIdOrLocationNotEmpty`?

File: src/main/scala/com/crimes/CrimeInvestigatorHelper.scala [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+    !(crime.crimeID.isEmpty ||
+      crime.latitude.isEmpty ||
+      crime.longitude.isEmpty)
```
Comment (azhur) [inline src/main/scala/com/crimes/CrimeInvestigatorHelper.scala]: imho `crime.crimeID.nonEmpty && crime.latitude.nonEmpty && crime.longitude.nonEmpty` is more readable

File: src/main/scala/com/crimes/model/Crime.scala [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+             val location: String,
+             val lSOACode: String,
+             val lSOAName: String,
+             val crimeType: String,
+             val lastOutcomeCategory: String,
+             val context: String
+           ) {
+  def this(csvLine: Array[String]) = this(csvLine(0), csvLine(1), csvLine(2), csvLine(3), csvLine(4), csvLine(5), csvLine(6), csvLine(7), csvLine(8), csvLine(9), csvLine(10), csvLine(11))
+
+  def this(crimeID: String, longitude: String, latitude: String) = this(crimeID, "", "", "", longitude, latitude, "", "", "", "", "", "")
```
Comment (azhur) [inline src/main/scala/com/crimes/model/Crime.scala]: for fields with possible empty or null values Option is a good fit.

File: src/main/scala/com/crimes/model/Crime.scala [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+
+  override def toString = s"Crime(CrimeID=$crimeID, Month=$month, ReportedBy=$reportedBy, FallsWithin=$fallsWithin, longitude=$longitude, latitude=$latitude, location=$location, lSOACode=$lSOACode, lSOAName=$lSOAName, CrimeType=$crimeType, lastOutcomeCategory=$lastOutcomeCategory, Context=$context)"
```
Comment (azhur) [inline src/main/scala/com/crimes/model/Crime.scala]: case class will do that automatically

File: src/main/scala/com/crimes/model/Crime.scala [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+}
```
Comment (azhur) [inline src/main/scala/com/crimes/model/Crime.scala]: scala has case class for this purpose. If you want more constructors you can do:

```scala
case class Crime(crimeID: String, month: String, ...)
object Crime {
  def apply(csvLine: Array[String]) = Crime(csvline(0), ...)
  def apply(rimeID: String, longitude: String, latitude: String) = ...
}
```

File: src/main/scala/com/crimes/model/CrimeStatistic.scala [hunk 7]
```diff
@@ -0,0 +1,25 @@
+package com.crimes.model
+
+class CrimeStatistic(
+                     val location: (String, String),
+                     val crimeTypes: List[String],
+                     val totalCrimes: Int
+                   ) extends Ordered[CrimeStatistic] {
```
Comment (azhur) [inline src/main/scala/com/crimes/model/CrimeStatistic.scala]: just fyi: usually in scala people don't extend Ordered. Sorting in scala collections works via implicits and typeclasses, no worries - we will cover this during the course.

File: src/main/scala/com/crimes/model/Crime.scala (hunk 4 above)
Comment (Nazariii) [reply src/main/scala/com/crimes/model/Crime.scala]: For me, it's not clear how Option will help with empty lines in this case as values always suppose to be empty and never null

File: src/main/scala/com/crimes/CrimeInvestigatorHelper.scala (hunk 3 above)
Comment (Nazariii) [reply src/main/scala/com/crimes/CrimeInvestigatorHelper.scala]: Yep, missed it, thanks.

File: src/main/scala/com/crimes/model/Crime.scala (hunk 4 above)
Comment (azhur) [reply src/main/scala/com/crimes/model/Crime.scala]: Empty lines should be skipped, Empty value could be parsed to None. As csv empty value logically means null.

**Pull request merged**